### PR TITLE
Metabase fixes

### PR DESCRIPTION
**Pull request by ei-grad**
I came to this changes into clickhouse-jdbc for https://github.com/metabase/metabase/pull/8722.

### CODE REVIEW

**Comment by ei-grad** [comment]
Its wrong, database name in ClickHouse is a FS directory name, and it looks like it could consist of any characters. Just use backtick quotes around the identifier as the ANSI SQL says for names escaping.

```use `test-data` ```

It works.

**Comment by enqueue** [comment]
You are right, escaping works fine, but the Metabase driver will need to know. Sorry for writing in this JDBC driver PR. I just checked it out, ran the Metabase test suite and encountered this error (using prostgres escaping mode).

**Comment by ei-grad** [comment]
No problems, you are welcome :).

File: src/main/java/ru/yandex/clickhouse/ClickHouseDatabaseMetadata.java [hunk 1]
```diff
@@ -703,7 +703,7 @@ public ResultSet getTables(String catalog, String schemaPattern, String tableNam
             String type, e = result.getString(3).intern();
             if (e == "View" || e == "MaterializedView" || e == "Merge" || e == "Distributed" || e == "Null") {
                 type = "VIEW"; // some kind of view
-            } else if (e == "Memory" || e == "Set" || e == "Join" || e == "Buffer") {
+            } else if (e == "Set" || e == "Join" || e == "Buffer") {
```
[the remark this answers is not in the document]
Comment (ei-grad) [reply src/main/java/ru/yandex/clickhouse/ClickHouseDatabaseMetadata.java]: Memory engine tables could be used in tests.

[the remark this answers is not in the document]
Comment (ei-grad) [reply src/main/java/ru/yandex/clickhouse/ClickHouseDatabaseMetadata.java]: Or like an alternative to the buffer tables in data pipeline.

**Comment by ei-grad** [comment]
> Is this still work in progress or can it be merged? I don't see a problem merging it now.

Nope, it would be great to merge it I think.

**Comment by serebrserg** [comment]
Sorry, I think I have broken your branch while resolving conflicts.
Resolved here https://github.com/yandex/clickhouse-jdbc/pull/285

**Pull request merged**